curses/client.cpp

```diff
@@ -561,29 +565,35 @@ int main(int argc, char * argv[])
 						break;
 
 					default:
+						display_needs_update = false;
 						//wmove(top_window, 2, 2);
 						//waddstr(top_window, to_string(c).c_str());
 						break;
 				}
 				if (isalnum(c))
 				{
+					display_needs_update = true;
 					if (jump_marks.find((char) c) != jump_marks.end())
 						index_of_first_visible_track = jump_marks[(char) c];
 				}
 			}
 			if (!keep_going)
 				break;
 
-			DACInfoCommand();
-			CurrentDACInfo();
-			TrackCount();
-			DisplayTracks();
-			wmove(top_window, 1, 1);
-			wrefresh(instruction_window);
-			wrefresh(top_window);
-			wrefresh(bottom_window);
-			wrefresh(mid_left);
-			wrefresh(mid_right);
+			// THESE ARE HAPPENING TOO FREQUENTLY.
+			if (display_needs_update || difftime(time(nullptr), last_update) > 0.2) {
+				DACInfoCommand();
```

Comment:
I would put all of these function calls into a wrapper function or something like `onUpdate`.